### PR TITLE
feat: re-support query engine execute dml

### LINKED ISSUE #2445: Support LogicalPlan::Insert to InsertRequest
### What type of enhancement is this?

Tech debt reduction

### What does the enhancement do?

Our logic of processing prepared statement is converting them into LogicalPlan with placeholder. Thus all prepared statement will eventually becomes a logical plan, including `INSERT INTO`.

But we only support insert with gRPC request and SQL AST. This ticket requests another way that support converting logical plan to insert request

### Implementation challenges

_No response_

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

* Remove calling to table.insert and table.delete, and replace with `TableMutationHandler`
* Clean some dead code

Hopefully, after this patch and #2482, #2445 should done

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/query/src/region_query.rs [hunk 1]
```diff
@@ -22,7 +22,6 @@ use crate::error::Result;
 
 #[async_trait]
 pub trait RegionQueryHandler: Send + Sync {
-    // TODO(ruihang): add trace id and span id in the request.
```
Comment (waynexia) [inline src/query/src/region_query.rs:25]: 🥇

File: src/operator/src/statement.rs [hunk 2]
```diff
@@ -104,14 +100,12 @@ impl StatementExecutor {
 
     pub async fn execute_sql(&self, stmt: Statement, query_ctx: QueryContextRef) -> Result<Output> {
         match stmt {
-            Statement::Query(_) | Statement::Explain(_) => {
+            Statement::Query(_) | Statement::Explain(_) | Statement::Delete(_) => {
                 self.plan_exec(QueryStatement::Sql(stmt), query_ctx).await
             }
 
             Statement::Insert(insert) => self.insert(insert, query_ctx).await,
```
Comment (waynexia) [inline src/operator/src/statement.rs:107]: Why `Insert` and `Delete` are on different paths

**Review by waynexia — APPROVED** [review]

File: src/operator/src/statement.rs (hunk 2 above)
Comment (zhongzc) [reply src/operator/src/statement.rs:107]: There is a fast path for insert. Delete doesn't have this fast path and can only use "plan + exec" like query.

File: src/query/src/table_mutation.rs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+use async_trait::async_trait;
+use session::context::QueryContextRef;
+use table::requests::{DeleteRequest, InsertRequest};
+
+use crate::error::Result;
+
+pub type AffectedRows = usize;
+
+#[async_trait]
```
Comment (killme2008) [inline src/query/src/table_mutation.rs:25]: Adds API docs for this trait and its functions.

**Review by killme2008 — COMMENTED** [review]
Good job! But  looks like this PR can't resolve #2445 .

**Comment by waynexia** [comment]
> Good job! But looks like this PR can't resolve #2445 .

needs both this and #2491 to fix

**Comment by killme2008** [comment]
> > Good job! But looks like this PR can't resolve #2445 .
> 
> needs both this and #2491 to fix

Cool,then we can add back the ignored integration and sqlness tests.

**Review by killme2008 — APPROVED** [review]
LGTM

**Pull request merged**